drivers/hmis/app/models/hmis/form/form_processor.rb

```diff
@@ -208,13 +208,17 @@ def exit_factory(create: true)
   def ce_assessment_factory(create: true)
     return owner if owner.is_a? Hmis::Hud::Assessment
 
-    self.ce_assessment ||= enrollment_factory.assessments.build(user_id: custom_assessment&.user_id) if create
+    return ce_assessment if ce_assessment.present? || !create
+
+    self.ce_assessment ||= enrollment_factory.assessments.build(user_id: custom_assessment&.user_id)
```

Comment:
Ah good find. The same issue is probably there with current living situation factory, too, since that is also one that is set up to be optionally collected on Custom Assessments. Can you fix that one as well?

Reply:
And, just to give some more context here... CurrentLivingSituation, CE Assessment, and CE Event were updated recently (at which point I introduced this bug) because they special in that they can be collected in 2 types of forms:
1. A form where the [Event] record is the form **owner**. In the interface, that would be collection under the CE 
Events tab on the enrollment dashboard.
1. A form where the [Event] record is a related record, and a Custom Assessment is the form **owner**. In the interface, that would be a Custom Assessment that collects a CE Event.